runtime/fail_byt.c

```diff
@@ -93,18 +93,14 @@ CAMLexport void caml_raise_with_string(value tag, char const *msg)
 */
 static void check_global_data(char const *exception_name)
 {
-  if (caml_global_data == 0) {
-    fprintf(stderr, "Fatal error: exception %s\n", exception_name);
-    exit(2);
-  }
+  if (caml_global_data == 0)
```

Comment:
This code is entered when a C function raises an exception and the bytecode runtime system is not initialized yet, i.e. the table of globals is not available.  This can happen in mixed C/OCaml programs that start in C then call into OCaml.  

The previous behavior was to treat this situation like an uncaught exception --- which it is, in a way: an exception is raised (from C) and no handler (in OCaml) is there to catch it.  Hence the error message and exit code were the same than those for an uncaught exception.  

Now you're turning this into an abort() with core dump.  This is not the same behavior.

Reply:
What you are saying somehow contradicts with the comment above this part of code: I thought the `caml_raise_xxx` functions were not meant to be used when the runtime is not initialized (like anything from the OCaml runtime, BTW). Rather, as the comment above suggests, I thought this check is needed if the `intern` function itself (which is responsible for decode global data) throws an exception.

If it is only about the exceptions raised by `intern`, then I would argue that `abort` is the right thing to do here. Indeed, the only I can see that would explain this happening in practice is a malformed bytecode executable (or under extreme memory constrains), which, in a way, is an internal error of the OCaml system (i.e., `ocamlc` produced a wrong executable).